WordPress/src/test/java/org/wordpress/android/ui/jetpack/backup/download/BackupDownloadViewModelTest.kt

```diff
@@ -1,30 +1,160 @@
+import org.wordpress.android.ui.jetpack.backup.download.BackupDownloadViewModel.ToolbarState.ProgressToolbarState
 import org.wordpress.android.util.wizard.WizardManager
+import org.wordpress.android.viewmodel.SingleLiveEvent
+import java.util.Date
 
-@RunWith(MockitoJUnitRunner::class)
-class BackupDownloadViewModelTest {
-    @Rule
-    @JvmField val rule = InstantTaskExecutorRule()
-
+@InternalCoroutinesApi
+class BackupDownloadViewModelTest : BaseUnitTest() {
     @Mock lateinit var wizardManager: WizardManager<BackupDownloadStep>
+    @Mock lateinit var backupDownloadStep: BackupDownloadStep
+    @Mock lateinit var savedInstanceState: Bundle
+    private val wizardManagerNavigatorLiveData = SingleLiveEvent<BackupDownloadStep>()
 
     private lateinit var viewModel: BackupDownloadViewModel
 
+    private val rewindId = "rewindId"
+    private val downloadId = 100L
+    private val published = Date(1609690147756)
+    private val url = "url"
+
+    private val backupDownloadState = BackupDownloadState(
+            activityId = "activityId",
+            rewindId = "rewindId",
+            downloadId = 100L,
+            siteId = 200L,
+            url = null,
+            published = Date(1609690147756)
+    )
+
     @Before
     fun setUp() {
+        whenever(wizardManager.navigatorLiveData).thenReturn(wizardManagerNavigatorLiveData)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = backupDownloadStep
+            Unit
+        }
         viewModel = BackupDownloadViewModel(wizardManager)
     }
 
+    @Test
+    fun `when view model is started, process moves to next step`() {
+        viewModel.start(null)
+
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, process moves to next step`() {
+        viewModel.start(null)
+        // need to clear invocations because nextStep is called on start
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when progress is finished, process moves to next step`() {
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+
+        viewModel.onBackupDownloadProgressFinished(url)
+        verify(wizardManager).showNextStep()
+    }
+
+    @Test
+    fun `when details is finished, state is updated properly`() {
+        val navigationTargets = initObservers().navigationTargets
+
+        viewModel.start(null)
+        clearInvocations(wizardManager)
+        whenever(wizardManager.showNextStep()).then {
+            wizardManagerNavigatorLiveData.value = BackupDownloadStep.PROGRESS
+            Unit
+        }
+
+        viewModel.onBackupDownloadDetailsFinished(rewindId, downloadId, published)
+
+        assertThat(navigationTargets.last().wizardState.rewindId).isEqualTo(rewindId)
```

Comment:
Minor (🔍): What do you think of the below alternative that utilised one `assertion` to achieve the same result?

```
assertThat(navigationTargets.last().wizardState)
                .isEqualTo(BackupDownloadState(rewindId = rewindId, downloadId = downloadId, published = published))
```

PS: Same applies to the below test. Although it already uses one `assertion` because it tried to verify only the `url` maybe it would be beneficial to keep them consistent. Wdyt?
PSS: Also, with this approach you are making sure that the `wizardState` is not being altered in any other way.

Reply:
Sounds good, thanks for the suggestion.